### ISSUE TITLE
HTTP Status 500 - Could not roll back Hibernate transaction

### ISSUE THREAD

**Issue by Otnehs** [dularion/streama#161]
After the server has been up for 12+ hours something seems to break, is there a requirement to run Oracle's Java or does Streama support OpenJDK? My server setup is currently Ubuntu 14.04.4 x64 and I'm currently utilizing OpenJDK 1.7.0_95 to run Streama. Everything works initially as far as I can tell but this has been a consistent issue where Tomcat will always start returning the following:

> 
> type Exception report
> 
> message Could not roll back Hibernate transaction; nested exception is org.hibernate.TransactionException: rollback failed
> 
> description The server encountered an internal error that prevented it from fulfilling this request.
> 
> exception
> 
> org.springframework.transaction.TransactionSystemException: Could not roll back Hibernate transaction; nested exception is org.hibernate.TransactionException: rollback failed
> 	org.springframework.orm.hibernate4.HibernateTransactionManager.doRollback(HibernateTransactionManager.java:581)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:849)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:826)
> 	org.springframework.transaction.support.TransactionTemplate.rollbackOnException(TransactionTemplate.java:164)
> 	org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:137)
> 	org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:85)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	streama.User.withTransaction(User.groovy)
> 	streama.User$withTransaction.call(Unknown Source)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	org.springframework.security.web.authentication.rememberme.TokenBasedRememberMeServices.processAutoLoginCookie(TokenBasedRememberMeServices.java:125)
> 	org.springframework.security.web.authentication.rememberme.AbstractRememberMeServices.autoLogin(AbstractRememberMeServices.java:115)
> 	org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:97)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:154)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:199)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192)
> 	org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 	org.codehaus.groovy.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:69)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.codehaus.groovy.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 
> root cause
> 
> org.hibernate.TransactionException: rollback failed
> 	org.hibernate.engine.transaction.spi.AbstractTransactionImpl.rollback(AbstractTransactionImpl.java:217)
> 	org.springframework.orm.hibernate4.HibernateTransactionManager.doRollback(HibernateTransactionManager.java:578)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:849)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:826)
> 	org.springframework.transaction.support.TransactionTemplate.rollbackOnException(TransactionTemplate.java:164)
> 	org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:137)
> 	org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:85)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	streama.User.withTransaction(User.groovy)
> 	streama.User$withTransaction.call(Unknown Source)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	org.springframework.security.web.authentication.rememberme.TokenBasedRememberMeServices.processAutoLoginCookie(TokenBasedRememberMeServices.java:125)
> 	org.springframework.security.web.authentication.rememberme.AbstractRememberMeServices.autoLogin(AbstractRememberMeServices.java:115)
> 	org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:97)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:154)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:199)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192)
> 	org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 	org.codehaus.groovy.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:69)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.codehaus.groovy.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 
> root cause
> 
> org.hibernate.TransactionException: unable to rollback against JDBC connection
> 	org.hibernate.engine.transaction.internal.jdbc.JdbcTransaction.doRollback(JdbcTransaction.java:167)
> 	org.hibernate.engine.transaction.spi.AbstractTransactionImpl.rollback(AbstractTransactionImpl.java:211)
> 	org.springframework.orm.hibernate4.HibernateTransactionManager.doRollback(HibernateTransactionManager.java:578)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:849)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:826)
> 	org.springframework.transaction.support.TransactionTemplate.rollbackOnException(TransactionTemplate.java:164)
> 	org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:137)
> 	org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:85)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	streama.User.withTransaction(User.groovy)
> 	streama.User$withTransaction.call(Unknown Source)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	org.springframework.security.web.authentication.rememberme.TokenBasedRememberMeServices.processAutoLoginCookie(TokenBasedRememberMeServices.java:125)
> 	org.springframework.security.web.authentication.rememberme.AbstractRememberMeServices.autoLogin(AbstractRememberMeServices.java:115)
> 	org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:97)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:154)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:199)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192)
> 	org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 	org.codehaus.groovy.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:69)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.codehaus.groovy.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 
> root cause
> 
> com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException: No operations allowed after connection closed.
> 	sun.reflect.GeneratedConstructorAccessor138.newInstance(Unknown Source)
> 	sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
> 	java.lang.reflect.Constructor.newInstance(Constructor.java:526)
> 	com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
> 	com.mysql.jdbc.Util.getInstance(Util.java:386)
> 	com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1015)
> 	com.mysql.jdbc.SQLError.createSQLException(SQLError.java:989)
> 	com.mysql.jdbc.SQLError.createSQLException(SQLError.java:975)
> 	com.mysql.jdbc.SQLError.createSQLException(SQLError.java:920)
> 	com.mysql.jdbc.ConnectionImpl.throwConnectionClosedException(ConnectionImpl.java:1320)
> 	com.mysql.jdbc.ConnectionImpl.checkClosed(ConnectionImpl.java:1312)
> 	com.mysql.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:5061)
> 	sun.reflect.GeneratedMethodAccessor654.invoke(Unknown Source)
> 	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	java.lang.reflect.Method.invoke(Method.java:606)
> 	org.apache.tomcat.jdbc.pool.ProxyConnection.invoke(ProxyConnection.java:126)
> 	org.apache.tomcat.jdbc.pool.JdbcInterceptor.invoke(JdbcInterceptor.java:109)
> 	org.apache.tomcat.jdbc.pool.DisposableConnectionFacade.invoke(DisposableConnectionFacade.java:80)
> 	com.sun.proxy.$Proxy34.rollback(Unknown Source)
> 	sun.reflect.GeneratedMethodAccessor654.invoke(Unknown Source)
> 	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	java.lang.reflect.Method.invoke(Method.java:606)
> 	org.springframework.jdbc.datasource.LazyConnectionDataSourceProxy$LazyConnectionInvocationHandler.invoke(LazyConnectionDataSourceProxy.java:376)
> 	com.sun.proxy.$Proxy36.rollback(Unknown Source)
> 	sun.reflect.GeneratedMethodAccessor654.invoke(Unknown Source)
> 	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	java.lang.reflect.Method.invoke(Method.java:606)
> 	org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy$TransactionAwareInvocationHandler.invoke(TransactionAwareDataSourceProxy.java:240)
> 	com.sun.proxy.$Proxy36.rollback(Unknown Source)
> 	org.hibernate.engine.transaction.internal.jdbc.JdbcTransaction.doRollback(JdbcTransaction.java:163)
> 	org.hibernate.engine.transaction.spi.AbstractTransactionImpl.rollback(AbstractTransactionImpl.java:211)
> 	org.springframework.orm.hibernate4.HibernateTransactionManager.doRollback(HibernateTransactionManager.java:578)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.processRollback(AbstractPlatformTransactionManager.java:849)
> 	org.springframework.transaction.support.AbstractPlatformTransactionManager.rollback(AbstractPlatformTransactionManager.java:826)
> 	org.springframework.transaction.support.TransactionTemplate.rollbackOnException(TransactionTemplate.java:164)
> 	org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:137)
> 	org.codehaus.groovy.grails.orm.support.GrailsTransactionTemplate.execute(GrailsTransactionTemplate.groovy:85)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	streama.User.withTransaction(User.groovy)
> 	streama.User$withTransaction.call(Unknown Source)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	org.springframework.security.web.authentication.rememberme.TokenBasedRememberMeServices.processAutoLoginCookie(TokenBasedRememberMeServices.java:125)
> 	org.springframework.security.web.authentication.rememberme.AbstractRememberMeServices.autoLogin(AbstractRememberMeServices.java:115)
> 	org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:97)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:154)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:199)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
> 	org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342)
> 	org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192)
> 	org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 	org.codehaus.groovy.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:69)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.codehaus.groovy.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:88)
> 	org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
> 	org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:344)
> 	org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:261)
> 
> root cause
> 
> com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: The last packet successfully received from the server was 42,218,935 milliseconds ago.  The last packet sent successfully to the server was 42,218,935 milliseconds ago. is longer than the server configured value of 'wait_timeout'. You should consider either expiring and/or testing connection validity before use in your application, increasing the server configured values for client timeouts, or using the Connector/J connection property 'autoReconnect=true' to avoid this problem.
> 	com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
> 	com.mysql.jdbc.SQLError.createCommunicationsException(SQLError.java:1129)
> 	com.mysql.jdbc.MysqlIO.send(MysqlIO.java:3988)
> 	com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2598)
> 	com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2778)
> 	com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2828)
> 	com.mysql.jdbc.ConnectionImpl.setAutoCommit(ConnectionImpl.java:5372)
> 	org.grails.datastore.gorm.GormStaticApi.findWhere(GormStaticApi.groovy:622)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService$_loadUserByUsername_closure1.doCall(GormUserDetailsService.groovy:64)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
> 	java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
> 	java.lang.Thread.run(Thread.java:745)
> 
> root cause
> 
> java.net.SocketException: Broken pipe
> 	java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113)
> 	java.net.SocketOutputStream.write(SocketOutputStream.java:159)
> 	java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
> 	java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
> 	com.mysql.jdbc.MysqlIO.send(MysqlIO.java:3969)
> 	com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2598)
> 	com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2778)
> 	com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2828)
> 	com.mysql.jdbc.ConnectionImpl.setAutoCommit(ConnectionImpl.java:5372)
> 	org.grails.datastore.gorm.GormStaticApi.findWhere(GormStaticApi.groovy:622)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService$_loadUserByUsername_closure1.doCall(GormUserDetailsService.groovy:64)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:815)
> 	org.grails.datastore.gorm.GormStaticApi.withTransaction(GormStaticApi.groovy:715)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:63)
> 	grails.plugin.springsecurity.userdetails.GormUserDetailsService.loadUserByUsername(GormUserDetailsService.groovy:81)
> 	grails.plugin.springsecurity.web.authentication.RequestHolderAuthenticationFilter.doFilter(RequestHolderAuthenticationFilter.java:49)
> 	grails.plugin.springsecurity.web.authentication.logout.MutableLogoutFilter.doFilter(MutableLogoutFilter.java:82)
> 	java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
> 	java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
> 	java.lang.Thread.run(Thread.java:745)
> 
> note The full stack trace of the root cause is available in the Apache Tomcat/7.0.55 logs.
>

**Comment by Otnehs** [comment]
I am going to try installing Oracle JDK and swap over the env vars to see if it makes any difference.

**Comment by Otnehs** [comment]
Swapping to Oracle JDK did not resolve the issue. Any ideas? I was looking into the settings in the DataSource.groovy file but it seems like the validationQuery should be keeping the session alive. Not sure if anyone else has experience this issue?

**Comment by dularion** [comment]
Hi! 
this seems to be an issue with your mysql server somehow. check out this message in your stacktrace: 

> com.mysql.jdbc.exceptions.jdbc4.CommunicationsException: The last packet successfully received from the server was 42,218,935 milliseconds ago. The last packet sent successfully to the server was 42,218,935 milliseconds ago. is longer than the server configured value of 'wait_timeout'. You should consider either expiring and/or testing connection validity before use in your application, increasing the server configured values for client timeouts, or using the Connector/J connection property 'autoReconnect=true' to avoid this problem.

maybe "autoReconnect=true" works for you?

I personally have a long-running instance of streama that has been running for approx 12 weeks without restart, and I don't have that issue... 
Btw, there should be no issue regarding oracleJDK or openjdk. I personally use openjdk.

**Comment by Otnehs** [comment]
When I ran it with "./grailsw run-war -Dgrails.env=production" it seems to be working correctly now, forcing the production environment seems to have resolved the issue as far as I can tell. Even though running without that command told me it was compiling the production version. Weird but now it has run for over 24 hours with zero issues (never did that before).

TL;DR: Use "./grailsw run-war -Dgrails.env=production"
